src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1796,14 +1812,14 @@ private void ScanBlockComment()
 
         // Implementation of the Levenshtein Distance algorithm
         // https://en.wikipedia.org/wiki/Levenshtein_distance
-        private static int GetStringSimilarity(string first, string second)
+        private static int GetStringSimilarity(string first, string second, int[,] distanceMap = null)
         {
             Diagnostics.Assert(!string.IsNullOrEmpty(first) && !string.IsNullOrEmpty(second), "Caller never calls us with empty strings");
 
             // Store a distance map to store the number of edits required to
             // convert the first <row> letters of First to the first <column>
             // letters of Second.
-            int[,] distanceMap = new int[first.Length + 1, second.Length + 1];
+            distanceMap ??= new int[first.Length + 1, second.Length + 1];
```

Comment:
Is this conditional allocation necessary?  It looks like distanceMap argument is always passed into this private method.

Reply:
This method could be used for other purposes, since itself is a general purpose string similarity calculation utility.

Reply:
Have you considered using the Lazy<> type here?

Reply:
Yes, considered it, but choose to go with the current implementation with two reasons:
1. Lazy<> requires a delegate for initialization. I'd like to avoid allocating the delegate when unnecessary.
2. The size of the 2-d array depends on the threshold defined here. I think it's better to make the initialization close to where the threshold is defined.